src/bufr/BufrParser/Exports/Variables/TimeoffsetVariable.cpp

```diff
@@ -0,0 +1,152 @@
+/*
+ * (C) Copyright 2022 NOAA/NWS/NCEP/EMC
+ *
+ * This software is licensed under the terms of the Apache Licence Version 2.0
+ * which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+ */
+
+#include <iomanip>
+#include <iostream>
+#include <locale>
+#include <ostream>
+#include <ctime>
+#include <vector>
+#include <unordered_map>
+
+#include "eckit/exception/Exceptions.h"
+#include "oops/util/Logger.h"
+
+#include "DataObject.h"
+#include "TimeoffsetVariable.h"
+#include "Transforms/TransformBuilder.h"
+
+
+namespace
+{
+    namespace ConfKeys
+    {
+        const char* Timeoffset = "timeOffset";
+        const char* Referencetime = "referenceTime";
+        const char* Transforms = "transforms";
+    }  // namespace ConfKeys
+}  // namespace
+
+
+namespace Ingester
+{
+    TimeoffsetVariable::TimeoffsetVariable(const std::string& exportName,
+                                           const std::string& groupByField,
+                                           const eckit::LocalConfiguration &conf) :
+      Variable(exportName),
+      groupByField_(groupByField),
+      conf_(conf)
+    {
+        initQueryMap();
+    }
+
+    std::shared_ptr<DataObjectBase> TimeoffsetVariable::exportData(const BufrDataMap& map)
+    {
+        checkKeys(map);
+
+        std::tm tm{};                // zero initialise
+        tm.tm_year = 1970 - 1900;    // 1970
+        tm.tm_mon = 0;               // Jan=0, Feb=1, ...
+        tm.tm_mday = 1;              // 1st
+        tm.tm_hour = 0;              // midnight
+        tm.tm_min = 0;
+        tm.tm_sec = 0;
+        tm.tm_isdst = 0;             // Not daylight saving
+        std::time_t epochDt = std::mktime(&tm);
```

Comment:
This is going to use local time unless you have set the environment variable TZ to "UTC" somewhere. We want everything in JEDI to be UTC. Check out how this is done in the chrono example in ioda engines: 

https://github.com/JCSDA-internal/ioda/blob/86f924e55fe4ce84f841820184b8480234cbb4c6/src/engines/Examples/Advanced/Complex_Data_Types/chrono.cpp#L64-L95

Reply:
Thank you Steve for catching this item.  I added it.

Reply:
I have learned about a better way to force UTC which is to replace mktime with another time.h function called timegm. This uses the tm struct as if it were UTC. I'll submit a PR to this branch with the changes to use timegm. I think this will be a better way to go since this works properly without having to force the TZ environment variable to UTC.

Reply:
PR has been submitted: #1112